docs/endpoint.md

```diff
@@ -5,14 +5,134 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
+
+### How to build endpoint application for wp77xx
+
+[Leaf](https://docs.legato.io/latest/toolsLeaf.html) is a workspace manager that will download, install and configure the required software packages for a Legato development environment.
+
+#### prerequisite packages required by leaf
+```
+$ sudo apt install \
+    python3-argcomplete \
+    python3-colorama \
+    python3-gnupg \
+    python3-jsonschema \
+    python3-requests \
+    gnupg \
+    bash-completion \
+    xz-utils
+```
+
+Install leaf
+```
+$ curl -sLO https://downloads.sierrawireless.com/tools/leaf/leaf_latest.deb && sudo dpkg -i leaf_latest.deb
+```
+
+Create a workspace
+```
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Setup the wp77xx target profile
+```
+$ leaf setup legato-latest -p swi-wp77_3.3.1
+```
+
+Finally, make the wp77xx endpoint target. Be careful with the directory of tangle-accelerator. It should be located within the workspace directory.
+```
+$ git clone https://github.com/DLTcollab/tangle-accelerator.git
+$ cd tangle-accelerator
+$ make TARGET=wp77xx legato # build endpoint as wp77xx target
+$ make TESTS=true TARGET=wp77xx legato # build endpoint as wp77xx target in test mode
+```
+
+### How to build endpoint application for native target
+
+Install required packages:
 ```
-$ bazel build //endpoint:wp7702
-$ bazel build //endpoint:sim
+$ sudo apt-get install -y   \
+    autoconf                \
+    automake                \
+    bash                    \
+    bc                      \
+    bison                   \
+    bsdiff                  \
+    build-essential         \
+    chrpath                 \
+    cmake                   \
+    cpio                    \
+    diffstat                \
+    flex                    \
+    gawk                    \
+    gcovr                   \
+    git                     \
+    gperf                   \
+    iputils-ping            \
+    libbz2-dev              \
+    libcurl4-gnutls-dev     \
+    libncurses5-dev         \
+    libncursesw5-dev        \
+    libsdl-dev              \
+    libssl-dev              \
+    libtool                 \
+    libxml2-utils           \
+    ninja-build             \
+    python                  \
+    python-git              \
+    python-jinja2           \
+    python-pkg-resources    \
+    python3                 \
+    texinfo                 \
+    unzip                   \
+    wget                    \
+    zlib1g-dev
 ```
 
+Create a workspace
+```
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Download legato by [git-repo](https://gerrit.googlesource.com/git-repo)
+```
+$ repo init -u git://github.com/legatoproject/manifest
+$ repo sync
+```
+
+You can also clone a specific version of `legato`
+```
+$ repo init -u git://github.com/legatoproject/manifest -m legato/releases/19.11.1.xml # specific legato 19.11.1 version
```

Comment:
You should address `19.11.1` as latest stable release Legato AF. How about [20.04](https://github.com/legatoproject/legato-af/commits/20.04-release)?

Reply:
Actually `19.11.1` and `20.04` are development version. The newest of stable version is version `19.07`. Should we change it?

Reply:
Address the preferable versions for Legato AF and explain.

Reply:
I think change to the stable version would be recommended.

Reply:
The current stable release is `20.04`. I will change it.

Reply:
Change back to the stable version `19.07`.